### PR TITLE
switching to another version if the geography does not exist in the selected version

### PR DESCRIPTION
**Pull request by emre2038**
## Description
switching to another version in which the selected geography exists, if the geography does not exist in the selected version

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-508

## How is it tested automatically?
added test steps into `__tests__/gui/multi_boundary_versions.feature`

## How should a reviewer test it locally
* navigate to a geography which does not exist in the current(or default) version. e.g. #geo:87ad36151ffffff in sandbox.youthexplorer
* confirm that the page is loaded successfuly
      - it switched to another version automatically
      - data mapper and the other elements are not stuck in the loading state

### Updated
* `src/js/versions/version_controller.js` to switch to another version

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally
- [x] 👩‍🎨 does the design match the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out
- [x] commit messages are meaningful

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers

### Testing

- [x]  ✅ added (appropriate) unit tests
- [x]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

File: src/js/versions/version_controller.js [hunk 1]
```diff
@@ -304,37 +314,58 @@ export class VersionController extends Component {
...
+        }
+        if (self._activeVersion === null || self._activeVersion === undefined || !self._activeVersion.model.exists) {
+            self._activeVersion = self.versions.filter((v) => {
+                return v.model.exists
+            })[0];
+        }
+
+        if (self._activeVersion === null || self._activeVersion === undefined) {
+            throw 'Selected geography does not exist in any of the versions'
+        }
```
Comment (michaelglenister) [inline src/js/versions/version_controller.js:347]: The if statements don't seem right to me e.g. In a case where `self._activeVersion === null` It will can only enter the first statement. The next if clauses will not be reached. Similar case for `self._activeVersion === undefined`

**Review by michaelglenister — COMMENTED** [review]
Added one comment, once is addressed this will be ready to move forward

File: src/js/versions/version_controller.js (hunk 1 above)
Comment (emre2038) [reply src/js/versions/version_controller.js:347]: * first `if` statement : if `self._activeVersion` is null, assign the version that is marked as default and is not 404
* second `if` statement : if `self._activeVersion` is still null, it means the default version is 404. so just assign the first version that is not 404
* third `if` statement : the selected geography does not exist in any of the versions. throw an error.

notice that those are not `else if`s - they are seperate `if`s. I don't see a problem here

Comment (emre2038) [reply src/js/versions/version_controller.js:347]: if you want to make sure, you can debug this code by navigating to geo:87ad36151ffffff in sandbox YE profile. since the default version is 404, you'd see that the second `if` is hit. 

if you go to a geography which does not exist like geo:random123 you will see the third `if` is hit

**Review by michaelglenister — APPROVED** [review]
